kafka/kafka.sh

```diff
@@ -24,49 +28,78 @@ function update_apt_get() {
   return 1
 }
 
-update_apt_get
-
-# Only run the installation on workers; verify zookeeper on master(s)
-if [[ $(/usr/share/google/get_metadata_value attributes/dataproc-role) == 'Master' ]]; then
-  service zookeeper-server status || \
-      (echo 'Required zookeeper-server not running on master!' && exit 1)
-  # On master nodes, just install kafka libs but not kafka-server.
-  apt-get install -y kafka
-  exit 0
-fi
-
-# Find zookeeper list first, before attempting any installation.
-ZOOKEEPER_CLIENT_PORT=$(grep clientPort /etc/zookeeper/conf/zoo.cfg | cut -d '=' -f 2)
-ZOOKEEPER_LIST=$(grep "^server\." /etc/zookeeper/conf/zoo.cfg | \
-        cut -d '=' -f 2 | cut -d ':' -f 1 | sed "s/$/:${ZOOKEEPER_CLIENT_PORT}/" | \
-        xargs echo | sed "s/ /,/g")
-if [[ -z "${ZOOKEEPER_LIST}" ]]; then
-  # Didn't find zookeeper quorum in zoo.cfg, but possibly workers just didn't
-  # bother to populate it. Check if YARN HA is configured.
-  ZOOKEEPER_LIST=$(bdconfig get_property_value --configuration_file \
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
```

Comment:
It appears that this method is almost always followed by exit code 1. Since we're using `set -e` above, consider adding `return 1` at the bottom of this function. That will cause the entire script to exit with code 1 on failure. If you still want to use `err` as a non-terminal function, consider adding the `return 1` variant as a new function `errAndFail` or similar.

Reply:
This perfectly makes sense to me to add return 1 at the bottom of err function, as I would like to terminate script after error.